pwnlib/elf/elf.py

```diff
@@ -852,6 +852,12 @@ def _patch_elf_and_read_maps(self):
             log.warn_once("Injected /proc/self/maps code did not execute correctly")
             return {}
 
+        # Sometimes the original binary already fail to run, for example,
+        # in case glibc mismatches.
+        if 'not found (required by' in data:
```

Comment:
since the valid format is `addr-addr rwx ...`. Maybe we could try to parse the if the first word is an valid hex address?